### PR TITLE
[Platformer] Type definitions and new tests on moving platforms

### PR DESCRIPTION
**Pull request by D8H**
This is changes with no effect. It will make the upcoming fix on the platform moving up easier to read.

### CODE REVIEW

File: Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js [hunk 1]
```diff
@@ -822,6 +846,108 @@ describe('gdjs.PlatformerObjectRuntimeBehavior', function () {
...
+      expect(object.getY()).to.be(-30);
+      expect(object.getX()).to.be(0);
+      expect(object.getBehavior('auto1').isOnFloor()).to.be(true);
+      expect(object.getBehavior('auto1').isFalling()).to.be(false);
+      expect(object.getBehavior('auto1').isMoving()).to.be(false);
+
+      // the platform is slightly overlapping the top of the object
+      platform.setY(object.getY() - platform.getHeight() + 1);
+      runtimeScene.renderAndStep(1000 / 60);
+      // Check that the object falls
```
Comment (4ian) [inline Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js:922]: Comment is outdated?

Comment (4ian) [reply Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js:922]: (should be "Check that the object follows the platform" I guess)

File: Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js [hunk 2]
```diff
@@ -822,6 +846,108 @@ describe('gdjs.PlatformerObjectRuntimeBehavior', function () {
...
+      expect(object.getBehavior('auto1').isOnFloor()).to.be(true);
+      expect(object.getBehavior('auto1').isFalling()).to.be(false);
+      expect(object.getBehavior('auto1').isMoving()).to.be(false);
+      expect(object.getY()).to.be(platform.getY() - object.getHeight());
+    });
+
+    it.skip('must not follow a platform that is moved over its top', function () {
```
Comment (4ian) [inline Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js:929]: This is skipped, whats happens currently?

**Review by 4ian — APPROVED** [review]
Added a few comments but looks good otherwise!

File: Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js (hunk 2 above)
Comment (D8H) [reply Extensions/PlatformBehavior/tests/platformerobjectruntimebehavior.spec.js:929]: It's put on top of the platform and stays on it. It does that as long as the platform is retrieved by the spacial search. The margin is maxFallingSpeed * timedelta, I guess.

**Comment by D8H** [comment]
I added a test for a regression I found in the fix I'm testing.

**Comment by 4ian** [comment]
Awesome, thank you!

**Pull request merged**